catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/AsyncFileAlterationObserver.java

```diff
@@ -66,6 +77,11 @@ public AsyncFileAlterationObserver(File fileToObserve, ObjectPersistentStore ser
     }
     this.serializer = serializer;
     rootFile = new AsyncFileEntry(fileToObserve);
+
+    if (LOGGER.isDebugEnabled()) {
+      timer = new Timer();
+      timer.scheduleAtFixedRate(new LogProcessing(), LOGGING_TIME_DELAY, LOGGING_TIME_INTERVAL);
```

Comment:
❗ This is actually not a good idea to do in the constructor as you are creating a timer and scheduling a task that will have an implicit reference to this object but this object has not yet been created.